src/encoded/static/components/genome_browser.js

```diff
@@ -61,21 +61,19 @@ function rAssemblyToSources(assembly, region) {
         for (let i = 0; i < positions.length; i += 1) {
             positions[i] = parseInt(positions[i].replace(/,/g, ''), 10);
         }
-        if (positions.length > 1) {
-            if (positions[0] > 10000) {
-                browserCfg.viewStart = positions[0] - 10000;
-            } else {
-                browserCfg.viewStart = 1;
-            }
-            browserCfg.viewEnd = positions[1] + 10000;
-        } else {
+        if (positions.length === 1) {
+            positions[1] = positions[0];
+        }
+        if ((positions[1] - positions[0]) < 10) {
             if (positions[0] > 10000) {
-                browserCfg.viewStart = positions[0] - 10000;
+                positions[0] -= 10000;
             } else {
-                browserCfg.viewStart = 1;
+                positions[0] = 1;
             }
-            browserCfg.viewEnd = positions[0] + 10000;
+            positions[1] += 10000;
         }
+        browserCfg.viewStart = positions[0];
+        browserCfg.viewEnd = positions[1];
         browserCfg.positionSet = true;
```

Comment:
I have no clue what this code is for, but that was true before this change as well. I know no one likes comments as I like comments, but I just have to trust this is fine, and hope I never have to modify it.

And I’ll be embarrassed if the original code was mine, which is possible.